ajax/merge.php

```diff
@@ -77,81 +78,114 @@
                 $plugin_sync_itemtype = 'PluginJamfMobileSync';
             }
 
-            $jamf_item = PluginJamfAPI::getMobileDeviceByID($jamf_id, true);
-            if ($jamf_item === null) {
-                // API error or device no longer exists in Jamf
-                throw new RuntimeException('Jamf API error or item no longer exists!');
```

Comment:
Without throwing the exception here, you aren't able to catch the case where the API doesn't return the data in the try/catch.